File: Microsoft Office Developer Documentation Team/SharePoint-2010-Creating-Visual-Web-Parts-in-4043/[VB]-SharePoint 2010 Creating Visual Web Parts in Visual Studio 2010/VB/InterviewSummary/VisualWebPart1/VisualWebPart1UserControl.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Linq
Imports System.Linq

Partial Public Class VisualWebPart1UserControl
    Inherits UserControl

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
        Dim thisWeb As SPWeb = SPContext.Current.Web
        Using ctx As EntitiesDataContext = New EntitiesDataContext("http://intranet.contoso.com")
            Try
                Dim query = From interviews In ctx.Interviews _
                            Select interviews.Title, interviews.Candidate.Name, interviews.Candidate.HomeCity, interviews.InterviewerImnName
                For Each Item In query
                    Dim interviewTreeNode As TreeNode = New TreeNode("Interview Details:" + Item.Title, Nothing, Nothing, thisWeb.Lists("Interviews").DefaultViewUrl, "_self")
                    Dim applicantTreeNode As TreeNode = New TreeNode("Applicant:" + Item.Name, Nothing, Nothing, thisWeb.Lists("Candidates").DefaultViewUrl, "_self")
                    Dim homecityTreeNode As TreeNode = New TreeNode("Home City:" + Item.HomeCity, Nothing, Nothing, thisWeb.Lists("Candidates").DefaultViewUrl, "_self")
                    Dim interviewerTreeNode As TreeNode = New TreeNode("Interviewer:" + Item.InterviewerImnName, Nothing, Nothing, thisWeb.Lists("Interviews").DefaultViewUrl, "_self")

                    interviewTreeNode.ChildNodes.Add(applicantTreeNode)
                    interviewTreeNode.ChildNodes.Add(homecityTreeNode)
                    interviewTreeNode.ChildNodes.Add(interviewerTreeNode)

                    TreeView1.Nodes.Add(interviewTreeNode)
                Next
            Catch ex As Exception
                TreeView1.Nodes.Add(New TreeNode("err" + ex.Message))
            End Try
        End Using
    End Sub

End Class

File: Microsoft Office Developer Documentation Team/SharePoint-2010-Creating-Visual-Web-Parts-in-4043/[C#]-SharePoint 2010 Creating Visual Web Parts in Visual Studio 2010/C#/InterviewSummary/VisualWebPart1/VisualWebPart1UserControl.ascx.cs

using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using Microsoft.SharePoint.Linq;
using System.Linq;
using Microsoft.SharePoint;


namespace InterviewSummary.VisualWebPart1
{
    public partial class VisualWebPart1UserControl : UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {

            SPWeb thisWeb = SPContext.Current.Web;

            using (EntitiesDataContext ctx = new EntitiesDataContext("http://intranet.contos.com"))
            {
                try
                {
                    var query = from interviews in ctx.Interviews
                                select new { interviews.Title, interviews.Candidate.Name, interviews.Candidate.HomeCity, interviews.InterviewerImnName };

                    foreach (var item in query)
                    {
                        TreeNode interviewTreeNode = new TreeNode("Interview Details:" + item.Title, null, null, thisWeb.Lists["Interviews"].DefaultViewUrl, "_self");
                        TreeNode applicantTreeNode = new TreeNode("Applicant:" + item.Name, null, null, thisWeb.Lists["Candidates"].DefaultViewUrl, "_self");
                        TreeNode homecityTreeNode = new TreeNode("Home City:" + item.HomeCity, null, null, thisWeb.Lists["Candidates"].DefaultViewUrl, "_self");
                        TreeNode interviewerTreeNode = new TreeNode("SplInterviewer:" + item.InterviewerImnName, null, null, thisWeb.Lists["Interviews"].DefaultViewUrl, "_self");

                        interviewTreeNode.ChildNodes.Add(applicantTreeNode);
                        interviewTreeNode.ChildNodes.Add(homecityTreeNode);
                        interviewTreeNode.ChildNodes.Add(interviewerTreeNode);
                        TreeView1.Nodes.Add(interviewTreeNode);

                    }
                }
                catch (Exception ex)
                {
                    TreeView1.Nodes.Add(new TreeNode("Err" + ex.Message));
                }
            }

        }
    }
}

